### ISSUE TITLE
Insert JSON in GUI an save real JSON 

### ISSUE THREAD

**Issue by Mattes1000** [jdorn/json-editor#355]
I got a property "custom", where the user of the JSON-Editor can insert "Custom" JSON. 
So I inserted 
```
<script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.1.8/ace.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.1.8/mode-json.js"></script>
```
and my schema looks

    {
  	"type" : "object",
	"properties" : {

		"custom" : {
			"type" : "object",
			"options" : {
				"disable_edit_json" : false
			}
		},
		"custom2" : {
			"type" : "string",
			"format" : "json"
		}
	  }
    }

Of course I get a string when I save the configuration. 

Is it possiple to parse the value of custom2 before saving?

Btw: It would be possible to let the user inser his JSON, but there is no "on the fly" validation..

**Comment by Z3roCoder** [comment]
You can Validate yourself what the user entered.

There are JavaScript functions like "JSON.parse(jObject)".

Just use the JSONEditor functions to get the content of property "custom2", parse it into JSON and save it into an extra variable. Then you can do whatever you want with it.To validate it immediatly there are several validators like TV4, zschema and more.

**Comment by Mattes1000** [comment]
But wouldn´t it be much nicer to write a custom editor inteface 

https://github.com/jdorn/json-editor#custom-editor-interfaces

Is this somwhow possible? Or are there any examples for that?

**Comment by Z3roCoder** [comment]
Yes there are examples. Just look into the existing code and look for the example he gave for the location. Dont neccesarily look for location but any of the existing ways and you can rather easily create a new interface

**Comment by jdorn** [comment]
If you have `type` set to `string`, then the value shouldn't be parsed into a JSON object.  The `format` keyword just specifies what type of string it should be. 

I think the editor you're describing (textarea/ace with the value parsed as JSON) should be used for schemas of type `object` or possibly ones with no type defined.  Something like this:
```json
{
  "type": "object",
  "additionalProperties": true,
  "title": "Freeform JSON"
}
```

The default object editor sucks for cases like this and a textarea would probably be better.  I might be able to implement this editor in the future.  I'd also be happy to accept pull requests for it.

**Comment by christx2** [comment]
Hey there, wanting to have a function to save the JSON of the populated attributes - can you suggest a function or some sample code, as I'd like to export the JSON to a file on the users desktop (Save Json button with a pop up the File Manager and where to save it). Thanks! Much appreciated.
